controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTester.java

```diff
@@ -263,7 +266,7 @@ private int countJobsOf(Application application) {
                                         .count();
     }
 
-    private static ApplicationPackage applicationPackage(String upgradePolicy) {
+    public static ApplicationPackage applicationPackage(String upgradePolicy) {
```

Comment:
Can we keep this private? I'd prefer tests explicitly building the application package they're going to use.

Reply:
But DeploymentTester is a tool to help _simplify_ testing, and this method is already used far and wide, just through indirections. Keeping this one method private won't help that. Also, I don't agree tests should always spell out all the irrelevant details.

Reply:
Only for upgrade tests. Most tests test deployment behavior of a specific application package, so exposing a "default one" with its details hidden away in some tester class just makes tests using it harder to reason about. Especially since tests usually assert on specific job types, which must match what's in the package.

Reply:
All these classes already use this method — most implicitly:

* DeploymentTriggerTest
* OutstandingChangeDeployerTest
* VersionStatusTest
* UpgraderTest

I say: let the default be available for those tests which _do not care_ about details.